flink-core/src/main/java/org/apache/flink/api/common/ExecutionConfig.java

```diff
@@ -148,6 +148,19 @@ public class ExecutionConfig implements Serializable, Archiveable<ArchivedExecut
                     .defaultValue(-1)
                     .withDescription(
                             "Should no longer be used because it is subsumed by RestartStrategyConfiguration");
+
+    /**
+     * Conceptually this is the same as {@link PipelineOptions#GLOBAL_JOB_PARAMETERS}, but it's
+     * stored/serialized in the configuration very differently. Use with care... First {@link
+     * PipelineOptions#GLOBAL_JOB_PARAMETERS} is read from the config, converted into {@link
+     * MapBasedJobParameters} and then its stored in the {@link #configuration} via {@link
+     * #GLOBAL_JOB_PARAMETERS_EXECUTION_CONFIG}. User can override this value via {@link
+     * ExecutionConfig#setGlobalJobParameters(GlobalJobParameters)}.
+     */
+    public static final ConfigOption<GlobalJobParameters> GLOBAL_JOB_PARAMETERS_EXECUTION_CONFIG =
+            key(PipelineOptions.GLOBAL_JOB_PARAMETERS.key())
+                    .defaultValue(new GlobalJobParameters());
```

Comment:
Ok. E2e test failure shows that this won't work that easily. `ConfigOption` doesn't support arbitrary types, so I would propose to store it as base64 serialised string.

Reply:
I guess this is also required for the Kryo serializers and the `restartStrategyConfiguration`.

I nevertheless like the change in general though.